src_features/signTx/logic_signTx.c

```diff
@@ -371,11 +380,7 @@ void finalizeParsing(bool direct) {
         io_seproxyhal_touch_tx_ok(NULL);
     } else {
         if (genericUI) {
-            ux_flow_init(
-                0,
-                ((dataPresent && !N_storage.contractDetails) ? ux_approval_tx_data_warning_flow
-                                                             : ux_approval_tx_flow),
-                NULL);
+            ux_approve_tx(dataPresent);
```

Comment:
Either we pass the bool `dataPresent`, or we could store that in a global.

Reply:
It looks good to me this way :+1: